pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -36,7 +39,11 @@ func (am *autoScalerManager) syncTiDB(tc *v1alpha1.TidbCluster, tac *v1alpha1.Ti
 		return nil
 	}
 	currentReplicas := tc.Spec.TiDB.Replicas
-	targetReplicas := calculateRecommendedReplicas(tac, v1alpha1.TiDBMemberType, client)
+	instances := filterTidbInstances(tc)
+	targetReplicas, err := calculateTidbMetrics(tac, sts, client, instances)
```

Comment:
This is the entrance function for TiDB to calculate auto-scaling by each metric.